domain/cached_repo_request_driven.py

```diff
@@ -48,13 +51,13 @@ def _update_cache_if_needed(self) -> None:
 
         logger.debug("Updating data cache start...")
         try:
-            results = self._source_repo.get_mesh_test_results(self._test_id, self._lookback_seconds)
+            results = self._source_repo.get_mesh_test_results(self._test_id, self._lookback_seconds, self._timeout)
             with self._cache_access_lock:
                 self._cache_test_results = results
                 self._cache_timestamp = datetime.now()
```

Comment:
I recommend to use UTC time because it does not have daylight saving and hence avoids problems with interval computation during switchover.
... and as I've mentioned in review to the other pull request, the timestamp should be in the `MeshResults` class.

Reply:
I agree. I will be better to do it in separate PR